compiler/noirc_evaluator/src/ssa/ssa_gen/context.rs

```diff
@@ -265,6 +266,190 @@ impl<'a> FunctionContext<'a> {
         Ok(self.builder.numeric_constant(value, typ))
     }
 
+    /// helper function which add instructions to the block computing the absolute value of the
+    /// given signed integer input. When the input is negative, we return its two complement, and itself when it is positive.
+    fn absolute_value_hlp(&mut self, input: ValueId, sign: ValueId, bit_size: u32) -> ValueId {
+        // We compute the absolute value of lhs
+        let one = self.builder.numeric_constant(FieldElement::one(), Type::bool());
+        let bit_width =
+            self.builder.numeric_constant(FieldElement::from(2_i128.pow(bit_size)), Type::field());
+        let sign_not = self.builder.insert_binary(one, BinaryOp::Sub, sign);
+        let as_field =
+            self.builder.insert_instruction(Instruction::Cast(input, Type::field()), None).first();
+        let sign_field =
+            self.builder.insert_instruction(Instruction::Cast(sign, Type::field()), None).first();
+        let positive_predicate = self.builder.insert_binary(sign_field, BinaryOp::Mul, as_field);
+        let two_complement = self.builder.insert_binary(bit_width, BinaryOp::Sub, as_field);
+        let sign_not_field = self
+            .builder
+            .insert_instruction(Instruction::Cast(sign_not, Type::field()), None)
+            .first();
+        let negative_predicate =
+            self.builder.insert_binary(sign_not_field, BinaryOp::Mul, two_complement);
+        self.builder.insert_binary(positive_predicate, BinaryOp::Add, negative_predicate)
+    }
+
+    fn check_overflow(
```

Comment:
Theres quite a bit of rightward shifting happening in this code thats making it harder to read, not a blocker though